Try to include <asm/page.h> only if __GLIBC__ is defined also.

--- a/unix/Xvnc/programs/Xserver/Xext/xf86bigfont.c	2014-07-20 02:45:59.000000000 +0200
+++ b/unix/Xvnc/programs/Xserver/Xext/xf86bigfont.c	2015-09-01 10:28:30.171432632 +0200
@@ -40,7 +40,7 @@
 
 #include <sys/types.h>
 #ifdef HAS_SHM
-#if defined(linux) && (!defined(__GNU_LIBRARY__) || __GNU_LIBRARY__ < 2)
+#if defined(linux) && (!defined(__GNU_LIBRARY__) || __GNU_LIBRARY__ < 2) && defined(__GLIBC__)
 /* libc4 does not define __GNU_LIBRARY__, libc5 defines __GNU_LIBRARY__ as 1 */
 /* Linux libc4 and libc5 only (because glibc doesn't include kernel headers):
    Linux 2.0.x and 2.2.x define SHMLBA as PAGE_SIZE, but forget to define
